subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/CompiledLanguage.groovy

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.fixtures;
+
+enum CompiledLanguage {
+    JAVA,
+    GROOVY;
+
+    String getName() {
+        return name().toLowerCase()
+    }
+
+    String getCompileTaskName() {
+        return "compile${name.capitalize()}"
+    }
+
+    String allProjectGroovyDependencies() {
+        this == GROOVY ?
+            """
+            allprojects {
+                dependencies {
+                    compile localGroovy()
```

Comment:
Should we do some cross-version tests with different Groovy versions instead of locking in the `localGroovy` distribution?

Reply:
we probably should, but I would like to make it in followups.